api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -72,16 +74,34 @@ func (mw *azureManagedMachinePoolWebhook) Default(ctx context.Context, obj runti
 	m.Labels[LabelAgentPoolMode] = m.Spec.Mode
 
 	if m.Spec.Name == nil || *m.Spec.Name == "" {
-		m.Spec.Name = &m.Name
+		aksNodePoolName := generateAKSNodePoolName(m.Name)
+		m.Spec.Name = &aksNodePoolName
```

Comment:
We can simplify the above two lines to `m.Spec.Name = ptr.To(generateAKSNodePoolName(m.Name))`

Reply:
Resolved @jackfrancis